pkg/doc/jwt/jwt.go

```diff
@@ -0,0 +1,457 @@
+/*
+ * Copyright SecureKey Technologies Inc. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package jwt
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+const (
+	// HeaderB64 defines b64 header of boolean type
+	HeaderB64 = "b64" // bool
+
+	// HeaderType defines typ header of string type
+	HeaderType = "typ" // string
+
+	// HeaderAlgorithm defines alg header of string type
+	HeaderAlgorithm = "alg" // string
+
+	// HeaderKeyID defines kid header of string type
+	HeaderKeyID = "kid" // string
+
+	// TypeJWT defines JWT type
+	TypeJWT = "JWT"
+
+	// AlgorithmNone used to indicate unsecured JWT
+	AlgorithmNone = "none"
+)
+
+const (
+	jwtPartsCount    = 3
+	jwtHeaderPart    = 0
+	jwtPayloadPart   = 1
+	jwtSignaturePart = 2
+)
+
+// JSONWebToken defines JWT basic operations with JWT headers, payload and signature.
```

Comment:
I do not understand why we are declaring interfaces and hidden structs in the same package. These interfaces seem unnecessary.